src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_apothecary_hummel.cpp

```diff
@@ -0,0 +1,577 @@
+            boss_apothecary_hummelAI(Creature* creature) : BossAI(creature, DATA_APOTHECARY_HUMMEL), _deadCount(0), _isDead(false) { }
+
+            void sGossipSelect(Player* player, uint32 menuId, uint32 gossipListId) override
+            {
+                me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+                CloseGossipMenuFor(player);
+                DoAction(ACTION_START_EVENT);
+            }
+
+            void Reset()
+            {
+                _Reset();
+                _deadCount = 0;
+                _isDead = false;
+                events.SetPhase(PHASE_ALL);
+                me->setFaction(FACTION_APOTHECARY_FRIENDLY);
+                me->SummonCreatureGroup(1);
+            }
+
+            void DoAction(int32 action)
+            {
+                if (action == ACTION_START_EVENT && events.IsInPhase(PHASE_ALL))
+                {
+                    events.SetPhase(PHASE_INTRO);
+                    events.ScheduleEvent(EVENT_HUMMEL_SAY_0, Milliseconds(1));
+
+                    me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                    me->setFaction(FACTION_APOTHECARY_HOSTILE);
+                    DummyEntryCheckPredicate pred;
+                    summons.DoAction(ACTION_START_EVENT, pred);
+                }
+            }
+
+            void DamageTaken(Unit* /*attacker*/, uint32& damage) override
+            {
+                if (damage >= me->GetHealth())
+                    if (_deadCount < 2)
+                    {
+                        damage = me->GetHealth() - 1;
+                        if (!_isDead)
+                        {
+                            _isDead = true;
+                            DoCastSelf(SPELL_PERMANENT_FEIGN_DEATH, true);
+                            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNK_29 | UNIT_FLAG_NOT_SELECTABLE);
+                            Talk(SAY_HUMMEL_DEATH);
+                        }
+                    }
+            }
+
+            void SummonedCreatureDies(Creature* summon, Unit* /*killer*/) override
+            {
+                if (summon->GetEntry() == NPC_APOTHECARY_FRYE || summon->GetEntry() == NPC_APOTHECARY_BAXTER)
+                    _deadCount++;
+
+                if (me->HasAura(SPELL_PERMANENT_FEIGN_DEATH) && _deadCount == 2)
+                    DoCastSelf(SPELL_QUIET_SUICIDE, true);
+            }
+
+            void JustDied(Unit* /*killer*/)
+            {
+                if (!_isDead)
+                    Talk(SAY_HUMMEL_DEATH);
+
+                events.Reset();
+                instance->SetBossState(DATA_APOTHECARY_HUMMEL, DONE);
+
+                Map::PlayerList const& players = me->GetMap()->GetPlayers();
+                if (!players.isEmpty())
+                {
+                    if (Group* group = players.begin()->GetSource()->GetGroup())
+                        if (group->isLFGGroup())
+                            sLFGMgr->FinishDungeon(group->GetGUID(), 288, me->GetMap());
```

Comment:
For me it would be better to add the 4 or 5 missing definitions in instance_encounters than keep doing this manual credit.
Lets see what the others think.